demo/react-hooks-controlled-multiple.html

```diff
@@ -49,64 +50,56 @@ <h2>Controlled stack</h2>
         // no need to init twice (would will return same grid) or register dup events
         const grid = gridRef.current = GridStack.init(
           {
-            float: true,
+            float: false,
             acceptWidgets: true,
             disableOneColumnMode: true, // side-by-side and fever columns to fit smaller screens
             column: 6,
             minRow: 1,
           },
-          `.controlled-${id}`
+          gridContainerRef.current
         )
-        .on('added', (ev, gsItems) => {
-          if (grid._ignoreCB) return;
-          // remove the new element as React will re-create it again (dup) once we add to the list or we get 2 of them with same ids but different DOM el!
-          // TODO: this is really not ideal - we shouldn't mix React templating with GS making it own edits as those get out of sync! see comment below @111.
-          gsItems.forEach(n => {
-            grid.removeWidget(n.el, true, false); // true=remove DOM, false=don't call use back!
-            // can't pass n directly even though similar structs as it has n.el.gridstackNode which gives JSON error for circular write.
-            addItem({id:n.id, x:n.x, y:n.y, w:n.w, h:n.h});
-          });
-        })
-        .on('removed', (ev, gsItems) => {
-          if (grid._ignoreCB) return;
-          gsItems.forEach(n => removeItem(n.id));
-        })
-
+          .on('added', (ev, gsItems) => {
+            if (grid._ignoreCB) return;
+            // remove the new element as React will re-create it again (dup) once we add to the list or we get 2 of them with same ids but different DOM el!
+            // TODO: this is really not ideal - we shouldn't mix React templating with GS making it own edits as those get out of sync! see comment below @111.
+            gsItems.forEach(n => {
+              grid.removeWidget(n.el, true, false); // true=remove DOM, false=don't call use back!
+              // can't pass n directly even though similar structs as it has n.el.gridstackNode which gives JSON error for circular write.
+              addItem && addItem({ id: n.id, x: n.x, y: n.y, w: n.w, h: n.h });
+            });
+          })
+          .on('removed', (ev, gsItems) => {
+            /* Looks like a bug in GridStack */
+            const dirtyNodes = grid.engine.getDirtyNodes();
+            if (dirtyNodes !== undefined && dirtyNodes.length !== 0) {
+              const newItems = grid.save(false);
+              changeItems && changeItems(newItems);
+            }
+
+            if (grid._ignoreCB) return;
+            gsItems.forEach(n => removeItem && removeItem(n.id));
+
+          })
+          .on('change', (ev, gsItems) => {
+            const newItems = grid.save(false);
```

Comment:
that is not the best way to do this... it's going to set a new list and REBUILD EACH widget all over again when only a few might have changed AND really only the x,y,w,h might have changed so the structure values need to change but not trigger a react rebuild because the items are already at the correct location. Only client side structure needs syncing.

Reply:
Do you mean creating a list with new object will force GridStack to rebuild each widget or React? Because as far as I know it is not the problem for React since ids which are use as keys were not changed hence React will not remount elements. Please correct me if i am wrong.

Reply:
did you actually debug react and made sure no new elements were created ?
I know in Angualr wrapper I did it doesn't as the id have not changed (what it keys off).

Reply:
Yes, I did.